metrics/cli/docs/02_metrics_cli_firebase_analytics_investigation.md

```diff
@@ -0,0 +1,80 @@
+# Motivation
+> What problem is this project solving?
+
+First of all, the Metrics project using Firebase Analytics to track useful information such as a user's login, a page's using, etc.
+
+Secondly, the Metrics CLI tool's goal is to deploy a Metrics project to Firebase from scratch, but for the newly created Firebase project, the Firebase Analytics is disabled. 
+
+Therefore, the document's goal is to investigate solutions of enabling Firebase analytics for a newly created Firebase project.
+
+## References
+> Link to supporting documentation, GitHub tickets, etc.
+
+- [Firebase Management API](https://firebase.google.com/docs/projects/api/reference/rest)
+- [Google Analytics Management API](https://developers.google.com/analytics/devguides/config/mgmt/v3/mgmtReference)
+- [Compute engine API](https://cloud.google.com/compute/docs/reference/rest/v1)
+- [Google OAuth 2.0](https://developers.google.com/identity/protocols/oauth2)
+- [gcloud CLI auth](https://cloud.google.com/sdk/gcloud/reference/auth)
+
+# Analyze
+
+## Process
+
+The analysis begins with the selection of the best solution to enable Firebase Analytics, which we investigated.
+
+The research concludes with a short description of the solutions, and a list of pros and cons of each.
+
+### Manual
+
+The first solution is to show a warning prompt to a user before the deploying hosting, which told him to enable Firebase Analytics in the Firebase console to make it work for the current project. Also, the warning prompt should contain a link to the Firebase Analytics page in the Firebase console.
+
+Let's describe the pros and cons of the manual solution.
+
+Pros:
+
+- easy to implement;
+
+- easy to do from the user side (visit the printed link and click the `Enable Google Analytics` button).
+
+Cons:
+
+- not an automatic solution.
+
+### Using an API
+
+The second solution is to use a [Firebase Management API](https://firebase.google.com/docs/projects/api/reference/rest) and a [Google Analytics Management API](https://developers.google.com/analytics/devguides/config/mgmt/v3/mgmtReference).
+
+The APIs contain the following required endpoints:
+- [`Google analytics summaries`](https://developers.google.com/analytics/devguides/config/mgmt/v3/mgmtReference/management/accountSummaries/list) to retrieve a Google analytics' account id;
+- [`Add google analytics`](https://firebase.google.com/docs/projects/api/reference/rest/v1beta1/projects/addGoogleAnalytics) to link the Google analytics with Firebase Analytics.
```

Comment:
Would it be great to explain the purpose of these APIs and why do we need them? Also, I believe it would be great to explain the whole process of automated enabling in a bit more detail to make it more understandable for readers.